core/src/KokkosExp_InterOp.hpp

```diff
@@ -75,35 +75,14 @@ using python_view_type_impl_t = typename python_view_type_impl<T...>::type;
 }  // namespace Kokkos
 
 namespace Kokkos {
-
-template <typename DataType, class... Properties>
-class DynRankView;
-
-namespace Impl {
-
-// Duplicate from the header file for DynRankView to avoid core depending on
-// containers.
-template <class>
-struct is_dyn_rank_view_dup : public std::false_type {};
-
-template <class D, class... P>
-struct is_dyn_rank_view_dup<Kokkos::DynRankView<D, P...>>
-    : public std::true_type {};
-
-}  // namespace Impl
-
 namespace Experimental {
 
 // ------------------------------------------------------------------ //
 //  this is used to extract the uniform type of a view
 //
 template <typename ViewT>
 struct python_view_type {
-  static_assert(
-      Kokkos::is_view<std::decay_t<ViewT>>::value ||
-          Kokkos::Impl::is_dyn_rank_view_dup<std::decay_t<ViewT>>::value,
-      "Error! python_view_type only supports Kokkos::View and "
-      "Kokkos::DynRankView");
+  static_assert(Kokkos::is_view<std::decay_t<ViewT>>::value);
```

Comment:
Did you mean to the drop the message?

Reply:
Yes, found it strange if the check doesn't match the message and recording all possible specializations here also doesn't seem ideal.
I don't feel strongly about it though and can it back if preferred.

Reply:
```suggestion
  static_assert(Kokkos::is_view<std::remove_reference_t<ViewT>>::value);
```
(because the `is_view` trait already works for `const`)

Reply:
I would not allow to change it as part of this PR.

Reply:
Got it! OK for me not to change it.